inc/saiswitch.h

```diff
@@ -325,6 +325,47 @@ typedef enum _sai_switch_failover_config_mode_t
 
 } sai_switch_failover_config_mode_t;
 
+/**
+ * @brief Defines tunnel encap ECN mode
+ */
+typedef enum _sai_switch_tunnel_encap_ecn_mode_t
+{
+    /**
+     * @brief Normal mode behavior defined in RFC 6040
+     * section 4.1 copy from inner
+     */
+    SAI_SWITCH_TUNNEL_ENCAP_ECN_MODE_STANDARD,
+
+    /**
+     * @brief User defined behavior
+     */
+    SAI_SWITCH_TUNNEL_ENCAP_ECN_MODE_USER_DEFINED
+
+} sai_switch_tunnel_encap_ecn_mode_t;
```

Comment:
I suggest to move
sai_tunnel_encap_ecn_mode_t and _sai_tunnel_decap_ecn_mode_t
from saitunnel.h to saiswitch.h
and use a single enum (similar to sai_packet_action_t, sai_packet_vlan_t, which are defined in saiswitch.h and also used in other header files)

Reply:
Won't it make it backward incompatible change? We can add comment for deprecation and plan it move it out in net major release.

Reply:
my suggestion is to move the enum as is, so it's backward compatible
not for example packet action, which is used in multiple headers -

    /** This is a combination of SAI packet action COPY_CANCEL and FORWARD */
    SAI_PACKET_ACTION_TRANSIT

} sai_packet_action_t;

the enum doesn't have to start with SAI_SWITCH, can be exactly as it is today

Reply:
I see what you are saying. Yes it makes sense. Committed the suggestion.

https://github.com/opencomputeproject/SAI/pull/1153/commits/f09a8fd1a14890e9d7ae126b47d171c8091fc2d9